test/external/external_test.go

```diff
@@ -75,13 +84,35 @@ var _ = Describe("External tests", func() {
 	})
 
 	Scenario("running chart move to intermediate bundle", func() {
+		start := time.Now()
+		steps.When("clear relok8s-save-cache")
 		steps.When(fmt.Sprintf("running relok8s chart move -y ../fixtures/testchart --image-patterns ../fixtures/testchart.images.yaml --to-intermediate-bundle %s/testchart-intermediate.tar", tmpDir))
 		steps.And("the move is computed")
 		steps.Then("the command says it will archive the chart")
 		steps.Then("the command says it is writing the hints file")
 		steps.Then("the command says it is writing the Helm Chart files")
 		steps.Then("the command says it is writing the container images")
 		steps.Then("the command says the intermediate bundle is complete")
+		steps.Then("relok8s-save-cache contains expected layer")
+		firstSaveDuration := time.Since(start)
+
+		start2 := time.Now()
+		info, err := os.Stat(expectedLayerFile())
+		Expect(err).ToNot(HaveOccurred())
+		modtime := info.ModTime()
+		steps.When(fmt.Sprintf("running relok8s chart move -y ../fixtures/testchart --image-patterns ../fixtures/testchart.images.yaml --to-intermediate-bundle %s/testchart-intermediate-2.tar", tmpDir))
+		steps.And("the move is computed")
+		steps.Then("the command says it will archive the chart")
+		steps.Then("the command says it is writing the hints file")
+		steps.Then("the command says it is writing the Helm Chart files")
+		steps.Then("the command says it is writing the container images")
+		steps.Then("the command says the intermediate bundle 2 is complete")
+		steps.Then("relok8s-save-cache contains expected layer")
+		secondSaveDuration := time.Since(start2)
+		info, err = os.Stat(expectedLayerFile())
+		Expect(err).ToNot(HaveOccurred())
+		Expect(info.ModTime()).To(Equal(modtime))
+		Expect(secondSaveDuration).To(BeNumerically("<", firstSaveDuration))
```

Comment:
This test case can be brittle, what if extracting the file from local takes more than the network speed? It should not be the case in general but you are adding a dependency here to the underlying system.

Reply:
timing checks removed

Reply:
> --- FAIL: TestGroupChangesByChart (0.00s)
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0
	
yep, i have it on my plate to take a look at it. I looked into it the other day but didn't manage to reproduce it

Reply:
It is very weird indeed, it happened a few times to me for a little while, then did not happen again for some time. It looks like some kind of race condition somewhere.